packages/react/src/components/Authenticator/shared/FormField.tsx

```diff
@@ -0,0 +1,61 @@
+import { FormField } from '@aws-amplify/ui';
+import {
+  PasswordField,
+  PhoneNumberField,
+  TextField,
+} from '../../../primitives';
+import { useAuthenticator } from '../hooks/useAuthenticator';
+
+export interface FormFieldProps {
+  name: string;
+  formFieldOptions: FormField;
+}
```

Comment:
Good point!

Reply:
I like the idea, and it'll work well especially when we refactor `formFields` into an array.

For now, since our `formFields` are organized in `{ [name: string]: FormFieldsOptions }`, I'll keep it as-is for that object key-value distinction. But I'll add a TODO comment here to put `name` into `FormFieldsOptions` for 3.0 refactor.

Reply:
TODO comment added here: https://github.com/aws-amplify/amplify-ui/pull/1484/commits/dee71f5eaad965606e65a80b82bc20e98fd336e9

Reply:
Talked offline, unresolving